incubator/crate/templates/statefulset.yaml

```diff
@@ -0,0 +1,98 @@
+apiVersion: "apps/v1"
+kind: StatefulSet
+metadata:
+  name: crate
+spec:
+  serviceName: "crate-db"
+  replicas: {{ .Values.crate.replicas }}
+  selector:
+    matchLabels:
+      app: {{ .Values.app }}
+  template:
+    metadata:
+      labels:
+        app: {{ .Values.app }}
+        app.kubernetes.io/name: crate-deployment
+        app.kubernetes.io/part-of: crate-db
+        app.kubernetes.io/version: {{ .Values.image.tag }}
+        app.kubernetes.io/managed-by: helm
+      annotations:
+        pod.alpha.kubernetes.io/initialized: "true"
+    spec:
+      initContainers:
+      - name: init-sysctl
+        image: busybox
+        imagePullPolicy: IfNotPresent
+        command: ["sysctl", "-w", "vm.max_map_count=262144"]
+        securityContext:
+          privileged: true
+      containers:
+      - name: crate
+        image: {{ .Values.image.name }}:{{ .Values.image.tag }}
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        command:
+          - /docker-entrypoint.sh
+          - -Clicense.enterprise=false
+          - -Ccluster.name={{ .Values.crate.clusterName }}
+          - -Cnetwork.host="0.0.0.0"
+          - -Cdiscover.seed_providers=srv
+          - -Cdiscovery.srv.query=_{{ .Values.service.clusterName }}._tcp.{{ .Values.service.name }}.{{ .Release.Namespace }}.svc.cluster.local
+          - -Cgateway.recover_after_nodes={{ .Values.crate.recoverAfterNodes }}
+          - -Cgateway.recover_after_time={{ .Values.crate.recoverAfterTime }}
+          - -Cgateway.expected_nodes={{ .Values.crate.replicas }}
+          - -Chttp.cors.enabled={{ .Values.http.cors.enabled }}
+          - -Chttp.cors.allow-origin={{ .Values.http.cors.allowOrigin }}
+          - -Cpath.data=/data
+          {{- if .Values.resources.limits.cpu }}
+          - -Cprocessors={{ .Values.resources.limits.cpu }}
```

Comment:
Would you mind showing the number of CPUs in the `values.yaml`?

Reply:
I don't think that's the right place or name for this value - I haven't used the Crate parameter but it sounds like it means `N` physical count? A k8s `limits` value of `1` would be a single hyperthread or vCPU, but it's often fractional, e.g. `900m` or `1.5` - would Crate accept that?

(Not to mention that the *limit* my not be what is actually allocated, in fact it probably isn't if it != `requests`, because this is happening on startup, not under load.)

Reply:
In CrateDB `processors` is used to calculate the size of threadpools. On bare metal or VMs the value is derived from the system, however inside Docker this value is not correct, because it reports the number of CPUs of the underlying host. That's why it must/should be overwritten when CrateDB is run in Docker.
As you stated correctly, it should be a positive integer, and cannot handle fractional values.

Reply:
I _suppose_ the correct value then is the ceil of `requests`, e.g. `900m -> 1`, `1.5 -> 2`. But I'm not really sure that's how it works, or whether `1.5` vCPUs as a k8s unit can be a share of any (larger than one) number of underlying vCPUs = hyperthreads.

Is underestimating, e.g. using `--Cprocessors=2 # ceil 1.5` when we actually have a 0.5 share of three vCPUs an issue?

Reply:
We also use `ceil` when we deploy our own clusters and have not seen any problems yet. However we usually also only use full CPU core values.